dicthash/dicthash.py

```diff
@@ -68,10 +67,9 @@ def _generate_string_from_dict(d, blacklist, whitelist, prefix=''):
         whitelist = d.keys()
 
     if blacklist is not None:
-        whitelist = [key for key in whitelist if key not in blacklist]
+        whitelist = (key for key in whitelist if key not in blacklist)
 
-    raw = [_unpack_value(d[key], prefix + unicode(key)) for key in sorted(whitelist)]
-    return ''.join(raw)
+    return ''.join(_unpack_value(d[key], prefix + unicode(key)) for key in sorted(whitelist))
```

Comment:
It is unrelated to the topic of the PR, but we should still change this:
`prefix` is a keyword argument to `_unpack_value` so we should use it like this. Thus, please insert a `prefix=prefix+...` here.